src/components/modals/UnknownTokenPrice/UnknownTokenPriceModal.vue

```diff
@@ -0,0 +1,113 @@
+<script setup lang="ts">
+import { computed, onBeforeMount, reactive } from 'vue';
+
+import TokenInput from '@/components/inputs/TokenInput/TokenInput.vue';
+
+import useTokens from '@/composables/useTokens';
+import usePoolCreation from '@/composables/pools/usePoolCreation';
+import { formatWordListAsSentence } from '@/lib/utils';
+
+type Props = {
+  isVisible: boolean;
+  unknownTokens: string[];
+};
+
+const props = withDefaults(defineProps<Props>(), {
+  isVisible: false,
+  unknownTokens: [] as string[]
+});
+
+const emit = defineEmits(['close']);
+
+/**
+ * STATE
+ */
+const unknownTokenPrices = reactive({});
+
+/**
+ * COMPOSABLES
+ */
+const { seedTokens } = usePoolCreation();
+const { tokens, injectPrices } = useTokens();
+
+/**
+ * LIFECYCLE
+ */
+onBeforeMount(() => {
+  for (const token of props.unknownTokens) {
+    unknownTokenPrices[token] = null;
+  }
+});
+
+/**
+ * COMPUTED
+ */
+const readableUnknownTokenSymbols = computed(() => {
+  const tokenSymbols = (props.unknownTokens || []).map(
+    tokenAddress => tokens.value[tokenAddress].symbol
+  );
+  return formatWordListAsSentence(tokenSymbols);
+});
+
+const isSubmitDisabled = computed(() => {
+  return props.unknownTokens.some(token =>
+    [null, ''].includes(unknownTokenPrices[token])
+  );
+});
+
+/**
+ * FUNCTIONS
+ */
+function getIndexOfUnknownToken(address: string) {
+  return seedTokens.value.findIndex(token => address === token.tokenAddress);
+}
+
+function injectUnknownPrices() {
+  injectPrices(unknownTokenPrices);
+  emit('close');
+}
+</script>
+
+<template>
+  <BalModal
+    title="Unknown token price"
+    :show="isVisible"
+    @close="$emit('close')"
+  >
+    <BalStack vertical isDynamic>
+      <p>
+        {{
+          $t('createAPool.unknownTokenPriceWarning', [
+            readableUnknownTokenSymbols
+          ])
+        }}
+      </p>
+
+      <span class="font-semibold">
+        {{ $t('createAPool.enterTokenPrice', [readableUnknownTokenSymbols]) }}
+      </span>
+      <BalStack isDynamic vertical>
+        <TokenInput
+          v-for="(address, i) in unknownTokens"
+          :key="i"
+          fixedToken
+          placeholder="$0.00"
+          v-model:amount="unknownTokenPrices[address]"
+          v-model:address="
+            seedTokens[getIndexOfUnknownToken(address)].tokenAddress
```

Comment:
Why is this needed instead of just passing `address`? Same with on line 101.

Reply:
It's not possible to bind address to a v-model since it comes from a iterated variable

Reply:
It's because you can't mutate props and unknownTokens is a prop. Because you are not modifying the address here you can just not use v-model, I think `:address="address"` will just work.